#MVSEC CONFIG FILE
EVENTS:
  # scene : /home/alessandro/datasets/mvsec/outdoor_day1/
  scene: "/home/alessandro/datasets/mvsec/indoor_flying/indoor_flying4_data/"
  side: "left"   #left 
  H: 260
  W: 346
  rectify: false
  use_valid_frame_range: true 



#CORNER DETECTOR PARAMETERS
FAST:
  threshold: 50
  nonmaxSuppression: false
  gradientScoring:
    enable: true
    desiredFeatures: 200

ACCUMULATOR:
  neutralPotential: 0.5
  eventContribution: 0.35
  decay: 1.0
  ignorePolarity: false

SLICING:
  type: "adaptive"    # "mvsec" | "fixed" | "adaptive"
  dt_ms: 10        # used only if type == "fixed"
  gt_mode : "dt1" #use "20hz", "dt1" or "dt4"


CAMERA:
  #camera intrinsic parameters
  fx : 196.73614846589038 
  fy : 196.5361311749205
  cx : 173.9835140415677
  cy : 134.63020721240977
  pixelSize: 0.0000185 #18.5 Î¼m from specifications (DAVIS346)

  


LK:
  winSize:
    width: 40  # Width of the window for Lucas-Kanade optical flow  (default: 21)
    height: 40  # Height of the window for Lucas-Kanade optical flow  (default: 21)

  maxLevel: 4  # Maximum pyramid level for Lucas-Kanade optical flow (default: 3)

  criteria:
    maxCount: 30  # Maximum number of iterations for Lucas-Kanade optical flow (default: 30)
    epsilon: 0.01  # Threshold for termination criteria of Lucas-Kanade optical flow (default: 0.01)

  flags: 0  # Flags for Lucas-Kanade optical flow 
  minEigThreshold: 0.001  # Threshold for minimum eigenvalue of Lucas-Kanade optical flow 


REJECTION_FILTER:
  magnitudeThresholdPixel: 10
  boundThreshold: 1.50          #threshold for rejecting the vectors magnitude. it must be > 1.0

#ADAPTIVE SLICING : 
adaptiveSlicing:
  enable: true
  P: 0.5
  I: 0.05
  D: 0.000
  maxTimingWindow: 50  #ms : min FPS = 20
  minTimingWindow: 10   #ms : max FPS = 100
  adaptiveTimingWindowStep: 1   #ms
  thresholdPIDEvents: 2
  OFPixelSetpoint: 3  #

#visualize image
visualizeImage: false
delayVisualize: 0